### ISSUE TITLE
Synchronize two or more timelines

### ISSUE THREAD

**Issue by 4integration** [almende/vis#2147]
Hi,

I am trying to get two (or more) timelines synchronized as it is @ http://almende.github.io/chap-links-library/js/timeline/examples/example10_link_two_timelines.html 

I can get the event and change the window but there is delay, i.e. it adjusts when stop the movement.
I can see the events being generated in log. 
How can it be solved?

      // Create a Timeline
      timeline2 = new vis.Timeline(container, items, groups, options);

	  timeline2.on('rangechange', function (properties) {
	  onrangechange2();
      logEvent('select', properties);
    });
    timeline2.on('rangechanged', function (properties) {
      onrangechange2();
      logEvent('select', properties);
	  
    });
    timeline2.on('select', function (properties) {
	  onrangechange2();
      logEvent('select', properties);
    });

.....

         function onrangechange1() {
            var range = timeline1.getWindow();
            timeline2.setWindow(range.start, range.end);
        }

        function onrangechange2() {
            var range = timeline2.getWindow();
            timeline1.setWindow(range.start, range.end);
        }
`

**Comment by 4integration** [comment]
[roadmap.zip](https://github.com/almende/vis/files/527121/roadmap.zip)

**Comment by 4integration** [comment]
Also if I impl the event listener on both it seems to get "circular dependency" and trigger each other

[roadmap.zip](https://github.com/almende/vis/files/527128/roadmap.zip)

**Comment by yotamberk** [comment]
Hi @4integration ,
The example you've attached doesn't contain the server to serve the files you're calling through AJAX in your client so the example doesn't work without some changes =(

After including the data explicitly in the html, everything seemed to work fine. The only thing is that there is an animation on the movement so it seems like the movement is registering only at the end of an action. Further exploration showed that it doesn't wait until the end of the action (try dragging but keeping the mouse down and see the dragging event triggering although the mouseup event hasn't finished). The animation makes it seem to be delayed.

In conclusion, the solution is to remove the timeline animation.
You can do this by adding `{animation: false}` on the setWindow functions: 
```javascript
timeline1.setWindow(range.start, range.end, {animation: false});
```

Does this help?

**Comment by 4integration** [comment]
It seems I missed some parts :( sorry about that.
Also fixed the data inclusion..

Two timelines, to be real-time synchronized when changed so listening on both "rangechange" and "rangechanged". This implies that if you change timeline1 it will notify timeline2 to move but then it seems that timeline2 send event "I moved" which notifies timeline1 to move and the battle is on :)

Attached updated source.
[roadmap1.zip](https://github.com/almende/vis/files/532573/roadmap1.zip)

**Comment by yotamberk** [comment]
It seems to work perfectly fine when setting
`timeline1.setWindow(range.start, range.end, {animation: false});`
I think the animation delay screws things up to cause some kind of loop calling one another and getting the weird behavior.
I suggest simply removing the animation with `{animation: false}`.

**Comment by 4integration** [comment]
Yes, it works! Thanks @yotamberk

**Comment by yotamberk** [comment]
No problem =)

**Comment by Reollyn** [comment]
When I call `rangechange` I'm getting this error `Error: Invalid start "NaN"`. Does anyone know why I'm getting that error?
